src/nvim/ui.c

```diff
@@ -171,6 +171,11 @@ void ui_refresh(void)
     pum_external &= ui->pum_external;
   }
 
+  if (updating_screen) {
+    loop_schedule(&main_loop, event_create(1, ui_refresh_event, 0));
+    return;
+  }
```

Comment:
`if` block could be moved before the declarations and `loop_schedule(...)` could be replaced with `ui_schedule_refresh()`.